tests/jobs.py

```diff
@@ -339,22 +341,79 @@ def test_backfill_max_limit_check_within_limit(self):
                           donot_pickle=True)
         job.run()
 
-        # dag run could not run since the max_active_runs has been reached
         dagruns = DagRun.find(dag_id=dag.dag_id)
-        self.assertEqual(4, len(dagruns))
+        self.assertEqual(2, len(dagruns))
         self.assertTrue(all([run.state == State.SUCCESS for run in dagruns]))
 
     def test_backfill_max_limit_check(self):
-        dag = self._get_dag_test_max_active_limits('test_backfill_max_limit_check')
+        dag_id = 'test_backfill_max_limit_check'
+        run_id = 'test_dagrun'
+        start_date = DEFAULT_DATE - datetime.timedelta(hours=1)
+        end_date = DEFAULT_DATE
+
+        def run_backfill():
+            dag = self._get_dag_test_max_active_limits(dag_id)
+
+            # this session object is different than the one in the main thread
+            thread_session = settings.Session()
+
+            # Existing dagrun that is not within the backfill range
+            dag.create_dagrun(run_id=run_id,
+                              state=State.RUNNING,
+                              execution_date=DEFAULT_DATE + datetime.timedelta(hours=1),
+                              start_date=DEFAULT_DATE)
+
+            thread_session.commit()
+
+            executor = TestExecutor(do_update=True)
+            job = BackfillJob(dag=dag,
+                              start_date=start_date,
+                              end_date=end_date,
+                              executor=executor,
+                              donot_pickle=True)
+            job.run()
+
+            thread_session.close()
+
+        backfill_job_thread = threading.Thread(target=run_backfill, name="run_backfill")
+        backfill_job_thread.start()
+
+        session = settings.Session()
+
+        # at this point backfill can't run since the max_active_runs has been reached,
+        # so it is waiting
+        dagruns = None
+        while not dagruns:
```

Comment:
Should have been a bit more clear sorry, the while loop is a problem too, typically this is solved by expecting some response and timing out if it takes more than X seconds (difference is it's a push, not a pull/poll). If it's too hard to set this up we can put in a sleep for now (ideally a smaller one, and have a timeout on the while loop to make sure the test suite doesn't crash/stall on this single test.